packages/ingest/HttpProviderClient.js

```diff
@@ -28,25 +29,45 @@ const validateHost = (host) => {
 
 class HttpProviderClient {
   constructor(providerConfig) {
-    this.requestOptions = {};
+    this.providerConfig = providerConfig;
     this.protocol = providerConfig.protocol;
     this.host = providerConfig.host;
     this.port = providerConfig.port;
+    this.gotOptions = { headers: { host: this.host } };
     this.certificateUri = providerConfig.certificateUri;
-
+    if (providerConfig.username && !providerConfig.password) {
+      throw new ReferenceError('Found providerConfig.username, but providerConfig.password is not defined');
+    }
     this.endpoint = buildURL({
       protocol: this.protocol,
       host: this.host,
       port: this.port
     });
   }
 
+  async setUpGotOptions() {
+    if (this.encrypted === true) {
+      this.username = await decrypt(this.providerConfig.username);
```

Comment:
I probably lack some fundamental understanding of basic auth: but how does this scenario work? A user configures encrypted (base64?) username/password for the provider, then we decrypt those credentials before putting them in the basic auth header? Is the point here just to avoid storing un-encrypted credentials on the provider and not having them exposed in the cumulus message?

Reply:
A user can provide the plaintext username/password when configuring the provider, but with `encrypted: true` that password will be encrypted when we create the provider record. So yes, this setting has us encrypt the credentials and store the encrypted copy to avoid exposing them in the record and message. Brings this ProviderClient in line with FTP/SFTP, which also support username/password.